packages/vscode-extension/src/commands/ImportFiles.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import * as path from 'path';
+import { QuickPickItem, window } from 'vscode';
+import ImportFileService from '../service/ImportFileService';
+import ListFilesService from '../service/ListFilesService';
+import { ANSWERS, ERRORS, IMPORT_FILES } from '../service/TranslationKeys';
+import { FolderItem } from '../types/FolderItem';
+import { actionResultStatus } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+
+const COMMAND_NAME = 'importfiles';
+
+export default class ImportFiles extends BaseAction {
+	private importFileService: ImportFileService;
+	private listFilesService: ListFilesService;
+
+	constructor() {
+		super(COMMAND_NAME);
+		this.importFileService = new ImportFileService(this.messageService);
+		this.listFilesService = new ListFilesService(this.messageService, this.translationService);
+	}
+
+	protected init(fsPath?: string) {
+		super.init(fsPath);
+		this.importFileService.setVsConsoleLogger(this.vsConsoleLogger);
+		this.listFilesService.setVsConsoleLogger(this.vsConsoleLogger);
+	}
+
+	protected async execute() {
+		if (!this.activeFile) {
+			// Already checked in validate
+			return;
+		}
+
+		const fileName = path.basename(this.activeFile, '.xml');
+
+		const selectedFilesPaths: string[] | undefined = await this.getSelectedFiles();
+		if (!selectedFilesPaths) {
+			return;
+		}
+
+		const excludeProperties = await window.showQuickPick(
+			[this.translationService.getMessage(ANSWERS.YES), this.translationService.getMessage(ANSWERS.NO)],
+			{
+				placeHolder: this.translationService.getMessage(IMPORT_FILES.QUESTIONS.EXCLUDE_PROPERTIES, fileName),
+				canPickMany: false,
+			}
+		);
+
+		const override = await window.showQuickPick(
+			[this.translationService.getMessage(ANSWERS.YES), this.translationService.getMessage(ANSWERS.NO)],
+			{
+				placeHolder:
+					selectedFilesPaths.length > 1
+						? this.translationService.getMessage(IMPORT_FILES.QUESTIONS.OVERRIDE)
+						: this.translationService.getMessage(IMPORT_FILES.QUESTIONS.OVERRIDE_SINGLE, selectedFilesPaths[0]),
+				canPickMany: false,
+			}
+		);
+
+		if (!override || override === this.translationService.getMessage(ANSWERS.NO)) {
+			this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_FILES.PROCESS_CANCELED));
+			return;
+		}
+
+		const destinationFolder = this.executionPath
+			? path
+					.dirname(this.activeFile)
+					.split(this.executionPath + '\\src')[1]
```

Comment:
This is making the code not compatible with old structure projects. We can use `this.getProjectFolderPath()` from BaseAction to get the projectFolder.